client/src/components/UploadProject.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from "react";
+import { Button } from "./Button";
+import {
+  Form,
+  LargeInput,
+  SmallInput,
+  Select,
+  ImageInput,
+} from "./InputElements";
+import { ImagePreview } from "./ImagePreview";
+import { useForm } from "react-hook-form";
+
+export default function UploadProject() {
+  const [imageInput, setImageInput] = useState("");
+  const [previewSource, setPreviewSource] = useState("");
+  const { register, handleSubmit } = useForm();
+
+  const handleImageChange = (event) => {
+    const file = event.target.files[0];
+    previewFile(file);
+    setImageInput(event.target.value);
+  };
+
+  const previewFile = (file) => {
+    const reader = new FileReader();
+    reader.readAsDataURL(file);
+    reader.onloadend = () => {
+      setPreviewSource(reader.result);
+    };
+  };
+
+  const onSubmit = (data) => {
+    const tags = data.tags.match(/[^,\s?]+/g);
+    console.log(data, tags);
+    if (!previewSource) return;
+    uploadImage(previewSource);
+  };
+
+  const uploadImage = async (base64EncodedImage) => {
+    try {
+      await fetch("/api/upload", {
+        method: "POST",
+        body: JSON.stringify({ data: base64EncodedImage }),
+        headers: { "Content-Type": "application/json" },
+      });
+      alert("Image successfully uploaded.");
```

Comment:
👀 Do you really want this ?

Reply:
the alert? for now yes . i'll focus on the functionality first and after that i'll implement nicer messages 👩🏻‍🎨

Reply:
okey accepted 😉